test/subspace-test-service/src/lib.rs

```diff
@@ -527,6 +527,23 @@ impl MockPrimaryNode {
             .await
     }
 
+    /// Remove all tx from the tx pool
+    pub fn clear_tx_pool(&self) -> Result<(), Box<dyn Error>> {
+        let txs: Vec<_> = self
+            .transaction_pool
+            .ready()
+            .map(|t| self.transaction_pool.hash_of(&t.data))
+            .collect();
+        self.transaction_pool
+            .pool()
+            .prune_known(&BlockId::Hash(self.client.info().best_hash), txs.as_slice())?;
+        self.transaction_pool
+            .pool()
+            .validated_pool()
+            .clear_stale(&BlockId::Number(self.client.info().best_number))?;
```

Comment:
This looks weird, since they are the same, why not

```rust
        let best_block_id = BlockId::Hash(self.client.info().best_hash);
        self.transaction_pool
            .pool()
            .prune_known(&best_block_id, txs.as_slice())?;
        self.transaction_pool
            .pool()
            .validated_pool()
            .clear_stale(&best_block_id)?;
```

I believe the simple code is preferable in general.